File: deliveryRoute/src/main/java/com/sparta/deliveryroute/infrastructure/client/HubFeignClient.java
package com.sparta.deliveryroute.infrastructure.client;

import com.sparta.deliveryroute.domain.client.HubClient;
import com.sparta.deliveryroute.infrastructure.client.response.RouteResult;
import com.sparta.deliveryroute.presentation.api.response.Response;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.UUID;

@FeignClient(
		name = "hub-server",
		fallbackFactory = HubFeignClientFallbackFactory.class
)
public interface HubFeignClient extends HubClient {

	@Override
	@GetMapping("/hubs/routes/id")
	Response<RouteResult> findHubRoutesById(@RequestParam UUID startHubId, @RequestParam UUID endHubId);

}

File: company/src/main/kotlin/com/sparta/company/infrastructure/client/HubFeignClient.kt
package com.sparta.company.infrastructure.client

import com.sparta.company.domain.client.HubClient
import com.sparta.company.presentation.api.response.Response
import io.github.resilience4j.retry.annotation.Retry
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import java.util.*

@FeignClient(
    name = "hub-server",
    fallbackFactory = HubFeignClientFallbackFactory::class
)
interface HubFeignClient : HubClient {

    @Retry(name = "hub")
    @GetMapping("/hubs/company/{hubId}")
    override fun existHub(
        @PathVariable hubId: UUID
    ): Response<Boolean>

}


